src/test/testRunner.ts

```diff
@@ -48,11 +48,17 @@ export function runTestsInFolder(testFolder: string): Promise<void> {
         const testFile = process.env['TEST_FILE'] === 'null' ? undefined : process.env['TEST_FILE']
         const testFilePath = testFile?.replace(/^src[\\\/]/, '')?.concat('.js')
 
-        const globalSetupPath = path.join(testsRoot, 'test', 'globalSetup.test.js')
-        if (testFilePath && fs.existsSync(globalSetupPath)) {
-            // XXX: explicitly add globalSetup, other tests depend on it.
-            mocha.addFile(globalSetupPath)
+        // Explicitly add globalSetup as the first test.
+        // TODO: migrate to mochaHooks (requires mocha 8.x).
+        // https://mochajs.org/#available-root-hooks
+        const globalSetupPath = path.join(testsRoot, 'test/globalSetup.test.js')
+        const globalSetupIntegPath = path.join(testsRoot, 'integrationTest/globalSetup.test.js')
+        if (!fs.existsSync(globalSetupPath) || !fs.existsSync(globalSetupIntegPath)) {
```

Comment:
only one of these is relevant for the method invoke, switching to path-as-a-parameter would tidy things here